sambacc/ctdb.py

```diff
@@ -586,6 +586,23 @@ def _convert_tdb_file(tdb_path: str, dest_dir: str, pnn: int = 0) -> None:
     subprocess.check_call(list(cmd))
 
 
+def archive_tdb(iconfig: config.InstanceConfig, dest_dir: str) -> None:
+    """Arhive TDB files into a given directory."""
+    # TODO: these paths should be based on our instance config, not hard coded
+    try:
+        os.mkdir(dest_dir)
+        _logger.debug("dest_dir: %r created", dest_dir)
+    except FileExistsError:
+        _logger.debug("dest_dir: %r already exists", dest_dir)
+    for tdbfile in _SRC_TDB_FILES:
+        for parent in _SRC_TDB_DIRS:
+            tdb_path = os.path.join(parent, tdbfile)
```

Comment:
Instead of double nested `for` loop, consider something like:

```
from itertools import product
from pathlib import Path

for tdb_path in [ Path(p) / f for p, f in product(_SRC_TDB_DIRS, _SRC_TDB_FILES) ]:
    ...
```

Reply:
This is how the older, migrate functions is written. I don't want to deviate from the pattern now.

Reply:
Fair enough